File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.java
import java.util.*;


class Solution {
    public String solution(String X, String Y) {
        String answer = "";
        StringBuilder sb = new StringBuilder();

        int[] XArr = new int[10];
        int[] YArr = new int[10];

        for(String s : X.split("")){
            int idx = Integer.parseInt(s);
            XArr[idx]++;
        }

        for(String s : Y.split("")){
            int idx = Integer.parseInt(s);
            YArr[idx]++;
        }

        int[] resultArr = new int[10];
        for(int i = 0; i < 10; i++){
            if(XArr[i] > YArr[i]){
                resultArr[i] = YArr[i];
            }else{
                resultArr[i] = XArr[i];
            }
        }

        for(int i = 9; i >= 0; i--){
            for(int j = 0; j < resultArr[i]; j++){
                sb.append(i);
            }
        }

        answer = sb.toString();

        if(answer.equals("")){
            answer = "-1";
        }else if(answer.replace("0", "") == ""){
            answer = "0";
        }

        return answer;
    }
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.swift
import Foundation

func solution(_ X:String, _ Y:String) -> String {
    var result = ""
    
    var xArr = [0,0,0,0,0,0,0,0,0,0]
    var yArr = [0,0,0,0,0,0,0,0,0,0]
    
    for i in X {
        if let num = Int(String(i)) {
            xArr[num] += 1
        }
    }
    
    for i in Y {
        if let num = Int(String(i)) {
            yArr[num] += 1
        }
    }
    
    var resultArr = Array<Int>()
    
    for i in 0...9 {
        if xArr[i] > yArr[i] {
            resultArr.append(yArr[i])
        } else {
            resultArr.append(xArr[i])
        }
    }
    
    for i in stride(from: 9, through: 0, by: -1) {
        for j in stride(from: 0, to: resultArr[i], by: 1) {
            result += String(i)
        }
    }
    
    if result == "" {
        result = "-1"
    }else if result.replacingOccurrences(of: "0", with: "") == "" {
        result = "0"
    }
    
    return result
}